### PR TITLE
[core][docs] Document `Property` component

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Closes https://linear.app/expo/issue/ENG-7152#comment-5e538794.
Adds docs section about `Property`

### CODE REVIEW

File: docs/pages/modules/module-api.mdx [hunk 1]
```diff
@@ -193,6 +193,71 @@ Events("onCameraReady", "onPictureSaved", "onBarCodeScanned")
 
 See [Sending events](#sending-events) to learn how to send events from the native code to JavaScript/TypeScript.
 
+</APIBox>
+<APIBox header="Property">
+
+Defines a value associated with a JavaScript object representing the native module.
+
+To declare a read-only property, you can use a shorthanded syntax that requires only two arguments:
```
Comment (amandeepmittal) [inline docs/pages/modules/module-api.mdx:201]:
```suggestion
To declare a read-only property, you can use a shorthanded syntax that requires two arguments:
```

File: docs/pages/modules/module-api.mdx [hunk 2]
```diff
@@ -193,6 +193,71 @@ Events("onCameraReady", "onPictureSaved", "onBarCodeScanned")
...
+
+```kotlin
+Property("foo") {
+  return@Property "bar"
+}
+```
+
+</CodeBlocksTable>
+
+In the case of the mutable property, both the getter and the setter closure are needed (using that syntax is also possible to declare a property with only a setter):
```
Comment (amandeepmittal) [inline docs/pages/modules/module-api.mdx:222]: I'm not sure I understand what is "that syntax" is being referenced in the above line?

**Review by amandeepmittal — COMMENTED** [review]
Thanks for documenting this! Left a small suggestion and a question around the wording that leads to the confusion.

File: docs/pages/modules/module-api.mdx [hunk 3]
```diff
@@ -193,6 +193,71 @@ Events("onCameraReady", "onPictureSaved", "onBarCodeScanned")
...
+}
+```
+
+</CodeBlocksTable>
+
+In the case of the mutable property, both the getter and the setter closure are needed (using the syntax below is also possible to declare a property with only a setter):
+
+- **name**: `String` — Name of the property that you'll use from JavaScript.
+- **getter**: `() -> ProperyType` — The closure to run when the getter for a property was called.
+- **setter**: `(newValue: ProperyType) -> void` — The closure to run when the setter for a property was called.
```
Comment (tsapeta) [inline docs/pages/modules/module-api.mdx:226]:
```suggestion
- **setter**: `(newValue: PropertyType) -> void` — The closure to run when the setter for a property was called.
```

File: docs/pages/modules/module-api.mdx [hunk 4]
```diff
@@ -196,7 +196,7 @@ See [Sending events](#sending-events) to learn how to send events from the nativ
 </APIBox>
 <APIBox header="Property">
 
-Defines a value associated with a JavaScript object representing the native module.
+Defines a new property directly on the JavaScript object that represents a native module, it is the same as calling [`Object.defineProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) on the module object.
```
Comment (amandeepmittal) [inline docs/pages/modules/module-api.mdx:199]:
```suggestion
Defines a new property directly on the JavaScript object that represents a native module. It is the same as calling [`Object.defineProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) on the module object.
```

I'd suggest breaking it into two sentences for readability 👍

**Review by amandeepmittal — APPROVED** [review]

**Pull request merged**